yt/utilities/answer_testing/utils.py

```diff
@@ -0,0 +1,459 @@
+            The io-friendly version of the passed iterable.
+    """
+    result = iterable.__class__.__name__ 
+    for elem in iterable:
+        # Check for user-defined functions
+        if inspect.isfunction(elem):
+            result += '_' + elem.__name__
+        # Non-string iterables (e.g., lists, tuples, etc.)
+        elif not isinstance(elem, str) and hasattr(elem, '__iter__'):
+            result += '_' + iterable_to_string(elem)
+        # Non-string non-iterables (ints, floats, etc.)
+        elif not isinstance(elem, str) and not hasattr(elem, '__iter__'):
+            result += '_' + str(elem)
+        # Strings
+        elif isinstance(elem, str):
+            result += '_' + elem
+    return result
+
+def hash_results(results):
+    r"""
+    Driver function for hashing the test result.
+
+    Parameters:
+    -----------
+        results : dict
+            Dictionary of {test_name : test_result} pairs.
+
+    Raises:
+    -------
+        None 
+
+    Returns:
+    --------
+        results : dict
+            Same as the passed results, but the test_results are now
+            hex digests of the hashed test_result.
+    """
+    # Here, results should be comprised of only the tests, not the test
+    # parameters
+    for test_name, test_value in results.items():
+        # These tests have issues with python-specific anchors and so
+        # are already hashed
+        # (see their definitions in yt/utilites/answer_testing/framework.py)
+        if test_name in ['projection_values', 'pixelized_projection_values', 'grid_values']:
+            continue
+        else:
+            results[test_name] = generate_hash(test_value)
+    return results
+
+def hash_dict(data):
+    r"""
+    Specifically handles hashing a dictionary object. 
+
+    Parameters:
+    -----------
+        data : dict
+            The dictionary to be hashed.
+
+    Raises:
+    -------
+        None 
+
+    Returns:
+    --------
+        hd.hexdigest : str
+            The hex digest of the hashed dictionary.
+    """
+    hd = None
+    for key, value in data.items():
+        if hd is None:
+            hd = hashlib.md5(bytes(key.encode('utf8')) + value.tobytes())
+        else:
+            hd.update(bytes(key.encode('utf8')) + value.tobytes())
+    return hd.hexdigest()
+
+def generate_hash(data):
+    r"""
+    Actually performs the hash operation. 
+
+    Parameters:
+    -----------
+        data : python object 
+            Data to be hashed.
+
+    Raises:
+    -------
+        None 
+
+    Returns:
+    --------
+        hd : str
+            Hex digest of the hashed data.
+    """
+    # Sometimes md5 complains that the data is not contiguous, so we
+    # make it so here
+    if isinstance(data, np.ndarray):
+        data = np.ascontiguousarray(data)
+    # Try to hash. Some tests return hashable types (like ndarrays) and
+    # others don't (such as dictionaries)
+    try:
+        hd = hashlib.md5(data).hexdigest()
```

Comment:
I haven't done any format benchmarking, but just from running it, it seems like the time to actually run the test is the dominant factor compared to actually performing the hashing. I did do some small testing with fnv_hash, though, and it is 1-1.2x faster on the numpy arrays I tried, so it could definitely still be worth going that route, but I wouldn't say that hashing is the bottleneck at the moment.

Reply:
If it’s only 20% faster to use fnv this is fine as-is.